lib/tx.js

```diff
@@ -16,15 +16,22 @@ function post (tx, pi) {
     .then(_.set('dirty', false))
 }
 
-function massage (tx, pi) {
+async function massage (tx, pi) {
   const direction = _.get('direction', tx)
   const cryptoCode = _.get('cryptoCode', tx)
+  const fiatCode = _.get('fiatCode', tx)
   const isDateField = r => r === 'created' || _.endsWith('_time', r)
   const transformDate = (v, k) => isDateField(k) ? new Date(v) : v
   const mapValuesWithKey = _.mapValues.convert({'cap': false})
   const transformDates = r => mapValuesWithKey(transformDate, r)
   const logCommission = r => _.assign(r, {
-    commissionPercentage: _.get(direction, pi.getCommissionPercentage(cryptoCode))
+    commissionPercentage: _.get(direction
+      , pi.getCommissionPercentage(cryptoCode))
+  })
+
+  const tickerPrice = await pi.getRawTickerPrice(fiatCode, cryptoCode)
```

Comment:
This is receiving proper error handling internally, but we should be careful on what we're adding on this method.

Reply:
Also this code is using async/await, is it ok?

Reply:
What part do we have to be careful about? I don't think it needs to be async/await. We should be caching current rates periodically and just use that cached value. We shouldn't poll each time we need the rates in the code.

Reply:
This `massage` function is a critical one, adding in async function calls without proper error handling can make tx fail for ignorable reasons (like saving ticker price). In this case we're safe because of the error handling inside the ticker.js

This is cached already, I'm just going to swap async await with promises